src/llfs/ioring_log_device2.cpp

```diff
@@ -0,0 +1,108 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/ioring_log_device2.hpp>
```

Comment:
Can we come up with a better name than `LogDevice2`? I think what we should probably do is delete all of the old LogDevice code and rename `LogDevice2` -> `LogDevice`. If `LogDevice2` is all around better than `LogDevice`, keeping `LogDevice` will just make our code base increase it's dead code footprint. The code for `LogDevice` will still be maintained of git if we ever need it. 

If there's anywhere that's still using `LogDevice` instead of `LogDevice2`, we should go ahead and replace it since we need to test `LogDevice2` as much as possible before release if that's what we plan on releasing.

Reply:
I agree about removing the old `IoRingLogDevice` code.  However, this PR is already too big, so maybe I can remove the old impl in a future PR?  As for selecting a new name vs renaming: it does look kind of corny, but I think it's a good thing that its an entirely new type, rather than a change to existing type, since it lets us track downstream breakage explicitly (and the on-disk formats are completely incompatible -- I don't want it to be in any way subtle when there is downstream usage that needs to be moved over).

Reply:
I think that makes a lot of sense and I'll agree to that work being moved into a future PR.